lib/http3/server.c

```diff
@@ -1370,14 +1373,19 @@ static int handle_input_expect_headers(struct st_h2o_http3_server_stream_t *stre
         }
         assert(stream->req.upgrade.base == NULL); /* otherwise PROTOCOL_EXISTS will be set */
         is_connect = 1;
-        expected_map |= H2O_HPACK_PARSE_HEADERS_SCHEME_EXISTS | H2O_HPACK_PARSE_HEADERS_PATH_EXISTS;
+        must_exist_map = H2O_HPACK_PARSE_HEADERS_METHOD_EXISTS | H2O_HPACK_PARSE_HEADERS_AUTHORITY_EXISTS |
+                         H2O_HPACK_PARSE_HEADERS_SCHEME_EXISTS | H2O_HPACK_PARSE_HEADERS_PATH_EXISTS;
+        may_exist_map = 0;
     } else {
         /* normal request */
-        expected_map |= H2O_HPACK_PARSE_HEADERS_SCHEME_EXISTS | H2O_HPACK_PARSE_HEADERS_PATH_EXISTS;
+        is_connect = 0;
+        must_exist_map = H2O_HPACK_PARSE_HEADERS_METHOD_EXISTS | H2O_HPACK_PARSE_HEADERS_AUTHORITY_EXISTS |
+                         H2O_HPACK_PARSE_HEADERS_SCHEME_EXISTS | H2O_HPACK_PARSE_HEADERS_PATH_EXISTS;
```

Comment:
I think we might want to move authority to "may".

It is a leftover from another issue which we fixed for HTTP/2 but forgot to do for HTTP/3 (see https://github.com/h2o/h2o/issues/3277), but as we are moving to the new design that correctly uses must's and may's, I think we might to fix the issue now.

Reply:
Done in f30c373.  Thank you!

I have amended the commit that is now b5feda3 so that it relaxes the `:scheme` check but only for the draft-mode CONNECT-UDP because that is the case that we need for supporting current "legacy" clients.